readthedocs/proxito/middleware.py

```diff
@@ -268,7 +272,7 @@ def process_request(self, request):  # noqa
             log.debug('Not processing Proxito middleware')
             return None
 
-        ret = map_host_to_project_slug(request)
+        ret = map_host_to_project(request)
```

Comment:
If we are expecting a `Response` or a `Project` object here, I'd change its return value to be a tuple: `(response, project)` where one of them will be `None`. That way, it will be pretty clear what's is what.

Reply:
I'm thinking of refactoring this, so it does the project extraction and response return in different steps, that way we don't need to mix the return types, the idea is making small refactors to get to that part.